Wikipedia/Code/NavigationBarHider.swift

```diff
@@ -32,23 +32,43 @@ public class NavigationBarHider: NSObject {
     
     fileprivate var isUserScrolling: Bool = false
     fileprivate var isScrollingToTop: Bool = false
+    fileprivate var isProgramaticallyScrolling: Bool = false
     var initialScrollY: CGFloat = 0
     var initialNavigationBarPercentHidden: CGFloat = 0
+
+    private func setDataForScroll(yOffset: CGFloat, topContentInset: CGFloat) {
+        guard let navigationBar = navigationBar else {
+            return
+        }
+        initialScrollY = yOffset + topContentInset
+        initialNavigationBarPercentHidden = navigationBar.navigationBarPercentHidden
+    }
+
+    public func setIsProgramaticallyScrolling(_ isProgramaticallyScrolling: Bool, yOffset: CGFloat, topContentInset: CGFloat) {
+        self.isProgramaticallyScrolling = isProgramaticallyScrolling
```

Comment:
Might want a `guard let isInteractiveHidingEnabled == true` here, just to confirm that the user has set the right navbar settings before you update these internal state properties.

Reply:
Good call, updated!